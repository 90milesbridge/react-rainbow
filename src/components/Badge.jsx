import React from 'react';
import classnames from 'classnames';

export default class Badge extends React.Component {
    render() {
        return (
            <span className={ this.getBadgeClass() }>
                { this.props.label }
            </span>
        )
    }

    getBadgeClass() {
        return classnames('slds-badge', this.props.className);
    }
}