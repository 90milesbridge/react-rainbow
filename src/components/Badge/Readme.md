Badge examples:

    const customBadgeStyles = {
        backgroundColor: '#2196f3',
        color: '#fff',
    };

    <div>
        <Badge label="Default Badge" />
        <Badge label="Inverse Badge" variant="inverse" />
        <Badge label="Custom Badge" style={customBadgeStyles} />
        <Badge label="Lightest Badge" variant="lightest" />
    </div>
