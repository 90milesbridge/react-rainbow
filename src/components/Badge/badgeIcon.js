import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../Icon';

function BadgeIcon(props) {
    const { isVisible, iconName, iconPosition, label } = props;

    const getIconClassName = () => {
        const hasLabel = label !== undefined && label !== null;
        return classnames({
            'slds-m-right_xx-small': hasLabel && iconPosition === 'left',
            'slds-m-left_xx-small': hasLabel && iconPosition === 'right',
        });
    };

    if (isVisible) {
        return <Icon size="xx-small" iconName={iconName} className={['slds-align_absolute-center', getIconClassName()]} />;
    }
    return null;
}

BadgeIcon.propTypes = {
    isVisible: PropTypes.bool,
    iconName: PropTypes.string,
    iconPosition: PropTypes.string,
    label: PropTypes.node,

};

BadgeIcon.defaultProps = {
    isVisible: false,
    iconName: null,
    iconPosition: null,
    label: null,
};

export default BadgeIcon;
