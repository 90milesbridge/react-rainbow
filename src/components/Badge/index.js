import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from './badgeIcon';

/**
 * Badges are labels which hold small amounts of information.
 */
export default function Badge(props) {
    const {
        className,
        style,
        label,
        variant,
        iconName,
        iconPosition,
    } = props;

    function getClassName() {
        return classnames('slds-badge', `slds-badge_${variant}`, className);
    }

    const showLeftIcon = iconName && iconPosition === 'left';
    const showRightIcon = iconName && iconPosition === 'right';

    return (
        <span className={getClassName()} style={style}>
            <Icon
                data-id="left-icon"
                iconName={iconName}
                iconPosition={iconPosition}
                label={label}
                isVisible={showLeftIcon} />
            {label}
            <Icon
                data-id="right-icon"
                iconName={iconName}
                iconPosition={iconPosition}
                label={label}
                isVisible={showRightIcon} />
        </span>
    );
}

Badge.propTypes = {
    /** The class name of the root element. */
    className: PropTypes.string,
    /** It is an object with custom style applied to the root element. */
    style: PropTypes.object,
    /** Label for the badge. */
    label: PropTypes.node,
    /** The badge variant. */
    variant: PropTypes.oneOf([
        'default', 'inverse', 'lightest',
    ]),
    /** The name of the icon. Names are written in the
     format '\utility:down\' where 'utility' is the category, and 'down' is the
     specific icon to be displayed. */
    iconName: PropTypes.string,
    /** The position of the icon if it is passed. */
    iconPosition: PropTypes.oneOf([
        'left', 'right',
    ]),
};

Badge.defaultProps = {
    className: undefined,
    style: {},
    variant: 'default',
    iconName: '',
    iconPosition: 'left',
    label: null,
};
