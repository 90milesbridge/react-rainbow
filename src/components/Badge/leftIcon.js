import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

export default function IconLeft({ iconName, position, label }) {
    if (!label && iconName && position === 'left') {
        return (
            <Icon iconName={iconName} size="xx-small" />
        );
    }
    if (label && iconName && position === 'left') {
        return (
            <Icon iconName={iconName} size="xx-small" className="slds-m-right_xx-small" />
        );
    }
    return null;
}

IconLeft.propTypes = {
    iconName: PropTypes.string,
    position: PropTypes.string,
    label: PropTypes.string,
};
IconLeft.defaultProps = {
    iconName: '',
    position: '',
    label: '',
};
