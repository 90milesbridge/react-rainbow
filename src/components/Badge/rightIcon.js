import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

export default function IconRight({ iconName, position, label }) {
    if (!label && iconName && position === 'right') {
        return (
            <Icon iconName={iconName} size="xx-small" />
        );
    }
    if (label && iconName && position === 'right') {
        return (
            <Icon iconName={iconName} size="xx-small" className="slds-m-left_xx-small" />
        );
    }
    return null;
}

IconRight.propTypes = {
    iconName: PropTypes.string,
    position: PropTypes.string,
    label: PropTypes.string,
};
IconRight.defaultProps = {
    iconName: '',
    position: '',
    label: '',
};
