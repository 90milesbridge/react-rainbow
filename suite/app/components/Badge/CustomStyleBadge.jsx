import React from 'react';
import { Badge } from './../../../../entry';
import styles from './styles.scss';

export default class CustomStyleBadge extends React.Component {
    render() {
        return (
            <div>
                <Badge label="Custom badge1" className="custom-badge"/>
            </div>
        )
    }
}