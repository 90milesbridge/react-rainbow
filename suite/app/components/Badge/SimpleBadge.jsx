import React from 'react';
import { Badge } from './../../../../entry';

export default class SimpleBadge extends React.Component {
    render() {
        return (
            <div>
                <Badge label="Badge label"/>
            </div>
        );
    }
}