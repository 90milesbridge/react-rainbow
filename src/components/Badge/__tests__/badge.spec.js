import React from 'react';
import renderer from 'react-test-renderer';
import Badge from './../index';

describe('<Badge/>', () => {
    it('should have the right class names when variant default and have a custom class', () => {
        const component = renderer.create(
            <Badge label="Badge label" className="custom-class" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when variant inverse', () => {
        const component = renderer.create(
            <Badge label="Inverse Badge" variant="inverse" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when variant lightest', () => {
        const component = renderer.create(
            <Badge label="Lightest Badge" variant="lightest" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when iconName is utility:world', () => {
        const component = renderer.create(
            <Badge iconName="utility:world" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when iconName is utility:world and iconPosition is right', () => {
        const component = renderer.create(
            <Badge iconName="utility:world" iconPosition="right" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when iconName is utility:world and iconPosition is left', () => {
        const component = renderer.create(
            <Badge iconName="utility:world" iconPosition="left" />,
        );
        expect(component).toMatchSnapshot();
    });
});
