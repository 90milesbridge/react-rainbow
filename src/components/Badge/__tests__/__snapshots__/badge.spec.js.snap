// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge/> should have the right class names when variant default and have a custom class 1`] = `
<span
  className="slds-align_absolute-center slds-badge slds-badge_default custom-class"
  style={Object {}}
>
  Badge label
</span>
`;

exports[`<Badge/> should have the right class names when variant inverse 1`] = `
<span
  className="slds-align_absolute-center slds-badge slds-badge_inverse"
  style={Object {}}
>
  Inverse Badge
</span>
`;

exports[`<Badge/> should have the right class names when variant lightest 1`] = `
<span
  className="slds-align_absolute-center slds-badge slds-badge_lightest"
  style={Object {}}
>
  Lightest Badge
</span>
`;
